tests/macros/utils.js

```diff
@@ -1,86 +1,243 @@
 };
+
+/**
+ * Asserts that value is false.
+ *
+ * @template T Type of value.
+ * @param {T} value Actual value.
+ */
 assert.isFalse = value => {
     expect(value).toEqual(false);
 };
+
+/**
+ * Asserts value is strictly greater than (>) floor.
+ *
+ * @param {number} value Actual value.
+ * @param {number} floor Minimum Potential expected value.
+ */
 assert.isAbove = (value, floor) => {
     expect(value).toBeGreaterThan(floor);
 };
+
+/**
+ * Asserts that value is an array.
+ *
+ * @template T Type of value.
+ * @param {T} value Actual value.
+ */
 assert.isArray = value => {
     expect(value).toBeInstanceOf(Array);
 };
+
+/**
+ * Asserts that value is an object of type 'Object'
+ * (as revealed by Object.prototype.toString).
+ *
+ * @template T Type of value.
+ * @param {T} value Actual value.
+ * @remarks The assertion does match subclassed objects.
+ */
 assert.isObject = value => {
     expect(value).toBeInstanceOf(Object);
 };
+
+/**
+ * Asserts that value is a function.
+ *
+ * @template T Type of value.
+ * @param {T} value Actual value.
+ */
 assert.isFunction = value => {
     expect(value).toBeInstanceOf(Function);
 };
+
+/**
+ * Asserts that object has a property named by property.
+ *
+ * @template T Type of object.
+ * @param {T} value Container object.
+ * @param {string} prop Potential contained property of object.
+ */
 assert.property = (value, prop) => {
     expect(value).toHaveProperty(prop);
 };
+
+/**
+ * Asserts that object doesn't have a property named by property.
+ *
+ * @template T Type of object.
+ * @param {T} value Container object.
+ * @param {string} prop Potential contained property of object.
+ */
 assert.notProperty = (value, prop) => {
     expect(value).not.toHaveProperty(prop);
 };
+
+/**
+ * Asserts that set1 and set2 have the same members. Order is not take into account.
+ *
+ * @template T Type of set values.
+ * @param {T[]} a1 Actual set of values.
+ * @param {T[]} a2 Potential expected set of values.
+ */
 assert.sameMembers = (a1, a2) => {
     expect(new Set(a1)).toEqual(new Set(a2));
 };
 
+/**
+ * Asserts that haystack includes needle.
+ *
+ * @template T Type of values in haystack.
+ * @param {string|T[]} list Container string or array.
+ * @param {string|T} element Potential value contained in haystack.
+ */
+assert.include = (list, element) => {
+    expect(list).toContain(element);
+};
+
+/**
+ * @param {string} macroName
+ * @return {Macro}
+ */
 function createMacroTestObject(macroName) {
     let templates = new Templates(__dirname + '/../../macros/');
     let pageContext = {
         locale: 'en-US',
         url: 'https://developer.mozilla.org/'
     };
     let environment = new Environment(pageContext, templates, true);
+    let ctx = environment.prototypeEnvironment;
+
+    /** @type {Map<string, string>} */
+    const macroResults = new Map();
+    const realTemplate = ctx.template;
+    ctx.template = jest.fn(async (name, ...args) => {
+        let macroName = String(name)
```

Comment:
Why add the String() conversion? Are there cases where this function is being invoked with a non-string argument?  If so I think we'd want that to cause a test failure rather than be accepted.